gen/build_deploy/bash.py

```diff
@@ -669,24 +668,31 @@ def fill_template(base_name, format_args):
     return installer_filename
 
 
-def make_dcos_launch():
-    # NOTE: this needs to be kept in sync with build_dcos_launch.sh
+def make_dcos_launch(variant, launch_info):
+    util_bootstrap_id = launch_info['bootstrap']
+    launch_package_id = None
+    for package_id in launch_info['packages']:
+        if pkgpanda.PackageId(package_id).name == 'dcos-launch':
+            launch_package_id = package_id
+            break
+    assert launch_package_id, 'dcos-launch package cannot be found!'
+    util_bootstrap_filename = util_bootstrap_id + '.bootstrap.tar.xz'
+    # Make this its own function for easy mocking
     work_dir = py.path.local.mkdtemp()
-    # the tree will not be in the working dir if run from downstream
-    tree_src = './'
-    if os.path.exists('ext/upstream'):
-        tree_src = './ext/upstream'
-    src_info = {'kind': 'git_local', 'rel_path': tree_src}
-    this_tree = GitLocalSrcFetcher(src_info, None, os.getcwd())
-    this_tree.checkout_to(str(work_dir))
-    # put the launch spec into the root of the tree before running pyinstaller
-    work_dir.join('dcos-launch.spec').write(pkg_resources.resource_string('launch', 'dcos-launch.spec'))
+    subprocess.check_call([
+        'cp', os.getcwd() + '/packages/cache/bootstrap/' + util_bootstrap_filename,
+        str(work_dir.join(util_bootstrap_filename))])
     with work_dir.as_cwd():
-        subprocess.check_call(['pyinstaller', '--log-level=DEBUG', 'dcos-launch.spec'])
-    subprocess.check_call(['mv', str(work_dir.join('dist').join('dcos-launch')), 'dcos-launch'])
-    work_dir.remove()
-
-    return "dcos-launch"
+        subprocess.check_call(['xz', '-d', util_bootstrap_filename])
+        subprocess.check_call(['tar', 'xf', util_bootstrap_filename.rstrip('.xz')])
+    prefix = 'packages/cache/'
+    if variant is not None:
+        launch_path = prefix + 'dcos-launch-{}'.format(pkgpanda.util.variant_name(variant))
+    else:
+        launch_path = prefix + 'dcos-launch'
+    subprocess.check_call([
```

Comment:
This is actually a change from https://github.com/dcos/dcos/pull/1353 and unrelated to the azure changes happening here. In any case, for what purpose would I catch this exception? I see no reason to here...